packages/mask/src/plugins/Trader/trader/useAllTradeComputed.ts

```diff
@@ -187,6 +187,18 @@ export function useAllTradeComputed(
     )
     const openoceanSwapEstimateGas = useDODOTradeGasLimit(openocean as TradeComputed<SwapRouteData> | null)
 
+    // traderjoe
+    const traderjoe_ = useUniswapV2Trade(
+        TradeProvider.TRADERJOE,
+        TradeStrategy.ExactIn,
+        inputAmount_,
+        '0',
+        tradeProviders.some((x) => x === TradeProvider.TRADERJOE) ? inputToken : undefined,
+        tradeProviders.some((x) => x === TradeProvider.TRADERJOE) ? outputToken : undefined,
```

Comment:
plz extract `tradeProviders.some((x) => x === TradeProvider.TRADERJOE)` as variable.

e.g:
```suggestion
        hasTraderJoeProvider ? inputToken : undefined,
        hasTraderJoeProvider ? outputToken : undefined,
```

Reply:
done